content/en/docs/concepts/extend-kubernetes/compute-storage-net/resource-drivers.md

```diff
@@ -0,0 +1,288 @@
+
+## Common calls for both allocation modes
+
+In both modes the allocation is preceded by getting parameters objects for ResourceClaim and
+ResourceClass to ensure the Resource Driver is able to get these objects and understand them.
+
+
+```gRPC
+    // GetClassParameters gets called to retrieve the parameter object
+    // referenced by a class. The content should be validated now if
+    // possible. class.Parameters may be nil.
+    //
+    // The caller will wrap the error to include the parameter reference.
+    GetClassParameters(ctx context.Context, class *resourcev1alpha2.ResourceClass) (interface{}, error)
+
+    // GetClaimParameters gets called to retrieve the parameter object
+    // referenced by a claim. The content should be validated now if
+    // possible. claim.Spec.Parameters may be nil.
+    //
+    // The caller will wrap the error to include the parameter reference.
+    GetClaimParameters(ctx context.Context, claim *resourcev1alpha2.ResourceClaim, class *resourcev1alpha2.ResourceClass, classParameters interface{}) (interface{}, error)
+
+    // Allocate gets called when all same-driver ResourceClaims for Pod are ready
+    // to be allocated. The selectedNode is empty for ResourceClaims with immediate
+    // allocation, in which case the resource driver decides itself where
+    // to allocate. If there is already an on-going allocation, the driver
+    // may finish it and ignore the new parameters or abort the on-going
+    // allocation and try again with the new parameters.
+    //
+    // Parameters have been retrieved earlier.
+    //
+    // Driver must set the result of allocation for every claim in "claims"
+    // parameter items. In case if there was no error encountered and allocation
+    // was successful - claims[i].Allocation field should be set. In case of
+    // particular claim allocation fail - respective item's claims[i].Error field
+    // should be set, in this case claims[i].Allocation will be ignored.
+    //
+    // If selectedNode is set, the driver must attempt to allocate for that
+    // node. If that is not possible, it must return an error. The
+    // controller will call UnsuitableNodes and pass the new information to
+    // the scheduler, which then will lead to selecting a different node
+    // if the current one is not suitable.
+    //
+    // The Claim, ClaimParameters, Class, ClassParameters fields of "claims" parameter
+    // items are read-only and must not be modified. This call must be idempotent.
+    Allocate(ctx context.Context, claims []*ClaimAllocation, selectedNode string)
+
+    // Deallocate gets called when a ResourceClaim is ready to be
+    // freed.
+    //
+    // The claim is read-only and must not be modified. This call must be
+    // idempotent. In particular it must not return an error when the claim
+    // is currently not allocated.
+    //
+    // Deallocate may get called when a previous allocation got
```

Comment:
If above "gets" words are changed to "is", then this should also be:
```suggestion
    // Deallocate may be called when a previous allocation is
```

Reply:
This is a quote from the code.

Reply:
Fixed.